packages/web3-eth-accounts/src/account.ts

```diff
@@ -53,10 +81,129 @@ export const privateKeyToAccount = (
 	const publicKeyString = `0x${publicKey.slice(2)}`;
 	const publicHash = sha3Raw(publicKeyString);
 	const publicHashHex = bytesToHex(publicHash);
-	const address = toChecksumAddress(publicHashHex.slice(-40)); // To get the address, take the last 20 bytes of the public hash
-	return { address, privateKey: stringPrivateKey, signTransaction, sign, encrypt };
+	return toChecksumAddress(publicHashHex.slice(-40)); // To get the address, take the last 20 bytes of the public hash
 };
 
+/**
+ * encrypt a private key given a password, returns a V3 JSON Keystore
+ * https://github.com/ethereum/wiki/wiki/Web3-Secret-Storage-Definition
+ */
+export const encrypt = async (
+	privateKey: HexString,
+	password: string | Buffer,
+	options?: CipherOptions,
+): Promise<Keystore> => {
+	if (!(isValidString(privateKey) || isBuffer(privateKey))) {
```

Comment:
```suggestion
	if (!isValidString(privateKey) || isBuffer(privateKey)) {
```
It looks like this method is expected `privateKey` to only be a `HexString`. We need to either change the if statement as above, or change the type to allow buffers and handle converting them via:

```typescript
const stringPrivateKey = Buffer.isBuffer(privateKey)
		? Buffer.from(privateKey).toString('hex')
		: privateKey;
```

I would suggest following the same logic as `privateKeyToAddress`:

```typescript
const privateKeyToAddress = (privateKey: string | Buffer): string => {
	if (!(isValidString(privateKey) || isBuffer(privateKey))) {
		throw new InvalidPrivateKeyError(privateKey);
	}

	const stringPrivateKey = Buffer.isBuffer(privateKey)
		? Buffer.from(privateKey).toString('hex')
		: privateKey;

	const stringPrivateKeyNoPrefix = stringPrivateKey.startsWith('0x')
		? stringPrivateKey.slice(2)
		: stringPrivateKey;

	// TODO Replace with isHexString32Bytes function in web3-eth PR:
	// Must be 64 hex characters
	if (stringPrivateKeyNoPrefix.length !== 64) {
		throw new PrivateKeyLengthError(stringPrivateKeyNoPrefix);
	}
// ...
```